File: Python/Homeworks/03/feladat.py
# Nev: Barna Gergely
# Neptun: FJKXGG
# h: h144988

def hogolyo_csata(list):
    temp = {}
    for kor in list:
        for ember, ertek in kor.items():
            if ember not in temp:
                temp[ember] = {'eldobott_hogolyok': 0, 'talalt': 0, 'fejtalalat': 0}

            for kulcs, ertek in ertek.items():
                if kulcs in temp[ember]:
                    temp[ember][kulcs] += ertek
                else:
                    temp[ember][kulcs] = ertek

    return temp


def hogolyo_pontossag(list):
    for i, e in list.items():
        e['pontossag'] = e.get("talalt") / e.get("eldobott_hogolyok")

    return list

File: JavaScript/Homeworks/03/feladat.js
// Nev: Barna Gergely
// Neptun: FJKXGG
// h: h144988

function clear(tomb) {
  if (tomb !== null && tomb !== undefined) {
    if (typeof tomb === "object" && tomb instanceof Array) {
      let forditottTomb = [];
      let ujtomb = [];
      for (const elem of tomb) {
        if (typeof elem === "number") {
          ujtomb.push(elem);
        }
        if (typeof elem === "string" && elem.length>0) {
          let szoveg = elem.substring(0, 2);
          szoveg = szoveg.toUpperCase();
          ujtomb.push(szoveg);
        }
      }
      for (let index = 0; index < ujtomb.length; index++) {
        forditottTomb[ujtomb.length-index-1] = ujtomb[index];
      }
      return forditottTomb;
    }
  } else {
    return 0;
  }
}

//console.log(clear([2, 5, 3, 'heyho', null, 7]));
